src/mscorlib/src/System/Guid.cs

```diff
@@ -1322,6 +1348,148 @@ unsafe private static int HexsToCharsHexOutput(char* guidChars, int a, int b)
+                offset += HexsToChars(guidChars + offset, _a >> 8, _a);
+                guidChars[offset++] = ',';
+                guidChars[offset++] = '0';
+                guidChars[offset++] = 'x';
+                offset += HexsToChars(guidChars + offset, _b >> 8, _b);
+                guidChars[offset++] = ',';
+                guidChars[offset++] = '0';
+                guidChars[offset++] = 'x';
+                offset += HexsToChars(guidChars + offset, _c >> 8, _c);
+                guidChars[offset++] = ',';
+                guidChars[offset++] = '{';
+                offset += HexsToCharsHexOutput(guidChars + offset, _d, _e);
+                guidChars[offset++] = ',';
+                offset += HexsToCharsHexOutput(guidChars + offset, _f, _g);
+                guidChars[offset++] = ',';
+                offset += HexsToCharsHexOutput(guidChars + offset, _h, _i);
+                guidChars[offset++] = ',';
+                offset += HexsToCharsHexOutput(guidChars + offset, _j, _k);
+                guidChars[offset++] = '}';
+            }
+            else
+            {
+                // [{|(]dddddddd[-]dddd[-]dddd[-]dddd[-]dddddddddddd[}|)]
+                offset += HexsToChars(guidChars + offset, _a >> 24, _a >> 16);
+                offset += HexsToChars(guidChars + offset, _a >> 8, _a);
+                if (dash)
+                    guidChars[offset++] = '-';
+                offset += HexsToChars(guidChars + offset, _b >> 8, _b);
+                if (dash)
+                    guidChars[offset++] = '-';
+                offset += HexsToChars(guidChars + offset, _c >> 8, _c);
+                if (dash)
+                    guidChars[offset++] = '-';
+                offset += HexsToChars(guidChars + offset, _d, _e);
+                if (dash)
+                    guidChars[offset++] = '-';
+                offset += HexsToChars(guidChars + offset, _f, _g);
+                offset += HexsToChars(guidChars + offset, _h, _i);
+                offset += HexsToChars(guidChars + offset, _j, _k);
+            }
+    }
+
+        // Returns whether the guid is successfully formatted as a span. 
+        public bool TryFormat(Span<char> destination, out int charsWritten, string format = null, IFormatProvider provider = null)
+        {
+            bool dash = true;
+            bool hex = false;
+            bool setBraces = false;
+            bool curly = false;
+            int offset = 0;
+            int guidSize;
+
+            if (format == null || format.Length == 0)
+                format = "D";
+
+            if (format.Length != 1)
+            {
+                // all acceptable format strings are of length 1
+                throw new FormatException(SR.Format_InvalidGuidFormatSpecification);
+            }
+
+            char formatCh = format[0];
+            if (formatCh == 'D' || formatCh == 'd')
+            {
+                guidSize = 36;
+            }
+
```

Comment:
I should have been more clear... my comment above about a blank line after the closing brace was for the standalone if block... no blank line is needed or desired before each else block, though.